### PR TITLE
v5 changelog

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Changelog for v5

- closes https://github.com/yoshuawuyts/nanomorph/issues/50
- closes https://github.com/yoshuawuyts/choo/issues/429

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+the state directly. In turn we've also removed the magic re-rendering and made
+it explicit. This enables people to create tight render loops that can even be
+used in GC constrained environments like games or music production. We think
+this change was well worth it, and will make a lot of sense going forward.
+
+People might also wonder why we've moved away from `flux`/`elm` and are using
+an event emitter now. It turns out that the previous architecture had a lot of
+confusing words that made it harder to learn than it should. It was also not
+possible to react to changes; the thing that changed always had to specify what
+needed to respond to it. By using event emitters we've inverted this, which
```
Comment (lrlna) [inline CHANGELOG.md]: imma pick on words, sorry 🙈 : maybe instead of `inverted this` just say `changed this`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+literally nothing left to remove from the API tho so this is probably the last
+time we get to break anything in a significant way._
+
+### changes
+- :exclamation: state is now mutable and renders are triggered through
+  `.emit('render')`.
+- :exclamation: we've replaced `.use()`, `.model()` and the rest of the choo
+  architecture with a reworked `.use()` method. It's called once on boot, and
+  exposes a mutable reference to `state` and an event emitter that's compatible
+  with Node's `require('events').EventEmitter`
```
Comment (lrlna) [inline CHANGELOG.md]: link to event emitter would be dope

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+- :exclamation: the `.router()` method has been replaced with `.route()`,
+  replacing the nested array API. This should be easier to remember and more
+  performant.
+- :exclamation: we've replaced `morphdom`/`yo-yo` with `nanomorph`. The two
+  algorithms are very comparable. The differences are that the new algorithm
+  is smaller and the value of input fields on re-rendering will be whatever the
+  `value=""` attribute is.
+- :exclamation: `choo/mount` is now available as `app.mount()`
```
Comment (sethvincent) [inline CHANGELOG.md]: maybe worth noting that `app.mount` calls `start` internally, so we'll use one or the other, right?

compared to v4 where `mount` was called using the tree returned from `app.start`

Comment (yoshuawuyts) [inline CHANGELOG.md]: Yeah good point!

**Pull request merged**